src/FactoryCollection.php

```diff
@@ -131,6 +146,12 @@ public static function sequence(Factory $factory, iterable $items): self
      */
     public function create(array|callable $attributes = []): array
     {
+        if ($this->isRootFactory && $this->factory instanceof PersistentObjectFactory && $this->factory->isPersisting()) {
+            return flush_after(
+               fn() => \array_map(static fn(Factory $f) => $f->create($attributes), $this->all())
+            );
+        }
```

Comment:
this allows to only flush once with `PersistentFactory::createMany()` and so on...